### PR TITLE
Add style & className props support

### PR DESCRIPTION
**Pull request by jaxxreal**
As a developer I need a possibility to style root element over usage `style` and `className` properties.

### CODE REVIEW

**Comment by jaxxreal** [comment]
Hi @ljharb! Please review and give me a feedback, when you have time! Thanks!

File: src/OutsideClickHandler.jsx [hunk 1]
```diff
@@ -16,7 +15,8 @@ const propTypes = forbidExtraProps({
   onOutsideClick: PropTypes.func.isRequired,
   disabled: PropTypes.bool,
   useCapture: PropTypes.bool,
-  display: PropTypes.oneOf(objectValues(DISPLAY)),
+  style: PropTypes.style,
+  className: PropTypes.string,
```
Comment (ljharb) [inline src/OutsideClickHandler.jsx]: We have very intentionally disallowed this antipattern; please see https://brigade.engineering/don-t-pass-css-classes-between-components-e9f7ab192785?gi=d97e4a1f93eb

**Comment by ljharb** [comment]
Similar to #12; #13, #8, #2.

**Comment by jaxxreal** [comment]
Thanks for feedback @ljharb! Looks like it worth checking closed PR's before submitting a new one. Sorry, my bad.

**Comment by ljharb** [comment]
No problem at all! Sorry we couldn't accommodate you.

**Comment by lencioni** [comment]
Do you think it would make sense to refactor this component to avoid rendering a DOM node and use forwardRef instead? (obviously this would be a breaking change)

**Comment by ljharb** [comment]
@lencioni i'm not sure how that would help if the wrapped component didn't directly render a DOM node - i don't think introducing a requirement that everything OutsideClickHandler wraps is either a DOM node or a forwardRef would be worth it.

**Comment by lencioni** [comment]
Yeah that makes sense. Perhaps it could be added as an option, so folks who want this type of control can opt in to it safely?

**Comment by ljharb** [comment]
I don't think the added complexity of such an option is really worth it - either way this is something we should discuss on a separate issue; it's off topic for this PR.

**Pull request closed without merging**